### PR TITLE
Validate commitment bridge transactions for finalized blocks

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

The PR introduces validation of bridge commitment transactions for finalized blocks as well. Verification includes checking quorum is reached and that signatures are valid.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Review by igorcrevar — APPROVED** [review]

File: consensus/ibft/consensus_backend.go [hunk 1]
```diff
@@ -215,21 +215,21 @@ func (i *backendIBFT) buildBlock(parent *types.Header) (*types.Block, error) {
...
 		Header:   header,
 		Txns:     txs,
 		Receipts: transition.Receipts(),
 	})
 
+	if err := i.PreCommitState(block, transition); err != nil {
+		return nil, err
+	}
+
+	_, root := transition.Commit()
```
Comment (goran-ethernal) [inline consensus/ibft/consensus_backend.go]: I don't think this is safe to move. I know it's IBFT, but in the `BuildBlock` function, we are calculating header hash, and after that we are setting `StateRoot` and `GasUsed` fields on header, which are part of the hash. Double check this.

Comment (Stefan-Ethernal) [reply consensus/ibft/consensus_backend.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1599/commits/336c9925926243cc7e2e03c8e90ef26dd7ceef3e

**Review by goran-ethernal — APPROVED** [review]

File: consensus/polybft/polybft.go [hunk 2]
```diff
@@ -656,8 +656,42 @@ func (p *Polybft) GetBlockCreator(h *types.Header) (types.Address, error) {
...
+			continue
+		}
+
+		decodedStateTx, err := decodeStateTransaction(tx.Input)
+		if err != nil {
+			return fmt.Errorf("unknown state transaction: tx=%v, error: %w", tx.Hash, err)
+		}
+
+		switch stateTxData := decodedStateTx.(type) {
+		case *CommitmentMessageSigned:
```
Comment (vcastellm) [inline consensus/polybft/polybft.go]: Should this be just an `if`?

**Review by vcastellm — APPROVED** [review]
Overall LGTM

File: consensus/polybft/polybft.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply consensus/polybft/polybft.go]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1599/commits/748e56e1740bb9988763efad499ccb76d97c87d8

**Pull request merged**